packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -150,6 +150,51 @@ class GestureRecognizerFactoryWithHandlers<T extends GestureRecognizer> extends
 /// ** See code in examples/api/lib/widgets/gesture_detector/gesture_detector.1.dart **
 /// {@end-tool}
 ///
+/// ## Troubleshooting
+///
+/// Why isn't my parent [GestureDetector] onTap method called ?
+///
+/// Given a parent [GestureDetector] which defines an onTap callback and a child
+/// [GestureDetector] that also defines an onTap callback. When the inner
+/// [GestureDetector] is tapped, both [GestureDetector]s enter the gesture arena
+/// because the pointer coordinates are in both [GestureDetector]s bounds. The child
+/// [GestureDetector] wins because it is in front.
+///
+/// Note that setting [GestureDetector.behavior] to [HitTestBehavior.opaque]
+/// or [HitTestBehavior.translucent] has no impact on parent-child relationships
+/// like in this case. To block hits on children use an [IgnorePointer] or [AbsorbPointer].
+///
+/// Note that some callbacks are called during the gesture arena resolution
```

Comment:
This section is a little confusing/misleading. In particular: `onTapDown` is always called (not might be called), I think, and if the detector looses in the arena `onTapCancel` will be called after the arena is already resolved.